### DISCUSSION TITLE
question about cudaMemcpy2D/3D

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#5595] [General]
Does SYCL provide similar functions for cudaMemcpy2D/3D  ? 

Thanks

**Comment by JackAKirk** [comment]
> Does SYCL provide similar functions for cudaMemcpy2D/3D ?
> 
> Thanks

Yes cuMemcpy2D/3D is used via the PI plugin for 2D/3D buffers and images.

**Reply by zjin-lcf** [reply]
Could you please add a link to PI plugin for 2D/3D buffers and images ?

**Reply by JackAKirk** [reply]
e.g. here: https://github.com/intel/llvm/blob/a618ca76e01a4be51dc27fdd8e1b85b0f805f203/sycl/plugins/cuda/pi_cuda.cpp#L2927

**Reply by zjin-lcf** [reply]
I suppose users would like to call functions in a SYCL program:

cgh.copy2D()
cgh.copy3D()
memcpy2D()
memcpy3D()

**Reply by keryell** [reply]
in the meantime you could implement yourself a user-defined C++ function `copy2D(q, ...)` that would call the right function through the SYCL interoperability mode of the CUDA bakend.

**Reply by zjin-lcf** [reply]
I tried to convert 2D memory copy to 1D memory copy. Thanks.
